### DISCUSSION TITLE
How to Convert Odoo Custom Modules into PIP Packages for CI/CD

### DISCUSSION THREAD

**Discussion by JayaramRangaraj** [OCA/odoo-community.org#187] [Q&A] [answered]
I am working on setting up CI/CD for my Odoo 17 instance, but I am facing difficulties in converting my custom Odoo modules into PIP packages. My goal is to be able to install these packages directly on my production server. Could anyone provide guidance on the steps I should follow to achieve this?

**Comment by yajo** [comment, marked as answer]
I think these days the easiest way is to package them with https://github.com/sbidoul/whool and publish them with https://twine.readthedocs.io/en/stable/. Check out the docs for both tools.

**Comment by rousseldenis** [comment]
> I am working on setting up CI/CD for my Odoo 17 instance, but I am facing difficulties in converting my custom Odoo modules into PIP packages. My goal is to be able to install these packages directly on my production server. Could anyone provide guidance on the steps I should follow to achieve this?

Is your question related to project custom addons (that should remain private) deployment ?

**Reply by JayaramRangaraj** [reply]
Yes
